### PR TITLE
(#788) Use recordcount instead of rowcount

### LINKED ISSUE #788: File upload record count is not correct
On file upload, record count is simply reported as the number of lines in the file, not the actual records. 

This should be fixed by getting the actual recordcount  instead of rowCount in the UploadedDataFile

To test:
- Upload a file for an existing instrument
- Check the 'No. of records' - stated for a file
- Store the file
- Check that the number of records agrees with previous stated

### PR DESCRIPTION
**Pull request by jonasfh**
Fixes #788

### CODE REVIEW

**Comment by squaregoldfish** [comment]
This is an interesting one to think about. There are several types of lines in files:

1. Headers
2. Blank lines
3. Measurements (can be split into marine, atmospheric etc)
4. Gas standards
5. Ignored lines (WAKE UP, SHUT DOWN etc)

But these only apply to files with a Run Type column. Other files only have headers, measurements and blank lines. What's the best thing to report to the user?

My guess for a relatively simple life right now is the number of data lines, i.e. don't count headers and blank lines. It should be easy to skip the headers since there are already functions for helping with that. We could do the more sophisticated stuff, but we should wait for user feedback and see if anyone cares.

File: WebApp/src/uk/ac/exeter/QuinCe/web/files/UploadedDataFile.java [hunk 1]
```diff
@@ -42,29 +41,11 @@ public UploadedFile getUploadedFile() {
   public void setUploadedFile(UploadedFile uploadedFile) {
     this.uploadedFile = uploadedFile;
     String[] fileLines = getLines();
-    rowCount = fileLines.length;
-    while ("".equals(fileLines[rowCount - 1])) {
```
Comment (squaregoldfish) [inline WebApp/src/uk/ac/exeter/QuinCe/web/files/UploadedDataFile.java]: There's a couple of other possible things to check:

* Lines that contain only whitespace
* Lines that contain only the separator character (i.e. every column value is blank)

I think we can treat both of these as empty lines. Whether the check should be here or in `getLines` I'm not sure.

Comment (jonasfh) [reply WebApp/src/uk/ac/exeter/QuinCe/web/files/UploadedDataFile.java]: Not sure if I understand: I'm removing all refs to rowCount in this commit, and rather using datafiles own recordCount - field. This should be correct. If not, that is another issue. Your comments seem to apply to the old rowCount - approach?

Comment (squaregoldfish) [reply WebApp/src/uk/ac/exeter/QuinCe/web/files/UploadedDataFile.java]: Oh, sorry. I got confused between this and the other PR. I shouldn't be doing this after filling my face with ibuprofen!

**Comment by squaregoldfish** [comment]
Merge!

**Pull request merged**